plutus-core/src/Language/PlutusCore/DeBruijn/Internal.hs

```diff
@@ -39,20 +40,25 @@ import           Data.Typeable
 
 import           Numeric.Natural
 
+import           Control.DeepSeq            (NFData)
 import           GHC.Generics
 
 -- | A relative index used for de Bruijn identifiers.
 newtype Index = Index Natural
```

Comment:
What's our policy on using `Natural`?  As far as I can see, the only place where we use these in `plutus-core` is here and in PLC version numbers.  It seems a bit odd that we use some very complicated type-level stuff to enforce invariants on data, but almost never use Natural in places where negative numbers should never occur.